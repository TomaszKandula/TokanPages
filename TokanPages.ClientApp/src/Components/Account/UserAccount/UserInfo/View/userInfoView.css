.backdrop {
    z-index: 9999 !important;
    color: #ffffff;
}

.user-id {
    margin-bottom: 30px;
}

@media (max-width: 600px) {
    .user-id {
        margin-bottom: 15px;
    }
}

.user-alias {
    margin-bottom: 30px;
}

@media (max-width: 600px) {
    .user-alias {
        margin-bottom: 15px;
    }
}

.user-email-status {
    margin-bottom: 4px;
}

@media (max-width: 600px) {
    .user-email-status {
        margin-bottom: 0;
    }
}

.user-email-verification {
    cursor: pointer;
    color: red;
    text-decoration: underline;
}

.user-avatar-text {
    margin-left: 15px !important;
}

.user-account-text-item {
    padding-top: 5px;
    padding-bottom: 5px;
}
