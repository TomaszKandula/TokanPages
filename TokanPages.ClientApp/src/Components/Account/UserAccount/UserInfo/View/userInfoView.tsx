import * as React from "react";
import Box from "@material-ui/core/Box";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Container from "@material-ui/core/Container";
import Skeleton from "@material-ui/lab/Skeleton";
import { UserMedia } from "../../../../../Shared/enums";
import { UploadUserMedia } from "../../../../../Shared/Components";
import { AccountFormInput } from "../../../../../Shared/Services/FormValidation";
import { ViewProperties } from "../../../../../Shared/interfaces";
import { ReactChangeEvent, ReactKeyboardEvent } from "../../../../../Shared/types";
import { UserInfoStyle, CustomSwitchStyle } from "./userInfoStyle";

import { 
    Button, 
    CircularProgress, 
    Divider, 
    Grid, 
    TextField, 
    Typography, 
    FormControlLabel 
} from "@material-ui/core";

import { 
    AuthenticateUserResultDto,
    SectionAccessDenied, 
    SectionAccountInformation, 
} from "../../../../../Api/Models";

interface BaseProperties extends ViewProperties
{    
    userStore: AuthenticateUserResultDto;
    accountForm: AccountFormInput;
    userImageName: string;
    isUserActivated: boolean;
    accountFormProgress: boolean;
    accountKeyHandler: (event: ReactKeyboardEvent) => void;
    accountFormHandler: (event: ReactChangeEvent) => void;
    accountSwitchHandler: (event: ReactChangeEvent) => void;
    accountButtonHandler: () => void;
    sectionAccessDenied: SectionAccessDenied;
    sectionAccountInformation: SectionAccountInformation;
}

interface Properties extends BaseProperties
{
    value: string;
}

const ReturnFileName = (value: string): string => 
{
    const maxFileNameLength: number = 8;
    const fileNameLength = value.length;
    const fileNameExtension = value.substring(fileNameLength - 3, fileNameLength);
    const shortFileName = value.substring(0, maxFileNameLength);

    return fileNameLength > maxFileNameLength ? `${shortFileName}~1.${fileNameExtension}` : value;
}

const CustomDivider = (args: { marginTop: number, marginBottom: number }): JSX.Element => 
{
    const classes = UserInfoStyle();
    return(
        <Box mt={args.marginTop} mb={args.marginBottom}>
            <Divider className={classes.divider} />
        </Box>
    );
}

const RenderText = (props: Properties): JSX.Element => 
{
    return props.isLoading ? <Skeleton variant="text" /> : <>{props.value}</>;
}

const UpdateAccountButton = (props: BaseProperties): JSX.Element => 
{
    const classes = UserInfoStyle();
    return(
        <Button 
            fullWidth 
            type="submit" 
            variant="contained" 
            onClick={props.accountButtonHandler} 
            disabled={props.accountFormProgress} 
            className={classes.button_update}>
            {!props.accountFormProgress 
            ? props.sectionAccountInformation?.updateButtonText 
            : <CircularProgress size={20} />}
        </Button>
    );
}

export const UserInfoView = (props: BaseProperties): JSX.Element => 
{
    const classes = UserInfoStyle();
    return(
        <section className={classes.section}>
            <Container maxWidth="md">
                <Box pt={15} pb={5}>
                    <Card elevation={0} className={classes.card}>
                        <CardContent className={classes.card_content}>
                            <Box pt={0} pb={0}>
                                <Typography component="span" className={classes.caption}>
                                    <RenderText 
                                        {...props} 
                                        value={props.sectionAccountInformation?.caption} 
                                    />
                                </Typography>
                            </Box>
                            <CustomDivider marginTop={2} marginBottom={1} />
                            <Box pt={5} pb={1}>
                                <Grid container spacing={2}>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelUserId} 
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        <Typography component="span" className={classes.user_id}>
                                            <RenderText 
                                                {...props} 
                                                value={props.userStore?.userId} 
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelUserAlias}
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        <Typography component="span" className={classes.user_alias}>
                                            <RenderText 
                                                {...props} 
                                                value={props.userStore?.aliasName} 
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelUserAvatar}
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        <Box className={classes.user_avatar_box}>
                                            {props.isLoading 
                                            ? null 
                                            : <UploadUserMedia mediaTarget={UserMedia.userImage} handle="userInfoSection_userImage" />}
                                            {props.isLoading 
                                            ? 
                                            null 
                                            : <Typography component="span" className={classes.user_avatar_text}>
                                                {ReturnFileName(props.userImageName)}
                                            </Typography>}
                                        </Box>
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelFirstName}
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" /> 
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="firstName"
                                            name="firstName" 
                                            variant="outlined" 
                                            value={props.accountForm?.firstName}
                                            onKeyUp={props.accountKeyHandler}
                                            onChange={props.accountFormHandler} 
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelLastName} 
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="lastName"
                                            name="lastName" 
                                            variant="outlined" 
                                            value={props.accountForm?.lastName}
                                            onKeyUp={props.accountKeyHandler}
                                            onChange={props.accountFormHandler} 
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelEmail}
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="email"
                                            name="email" 
                                            variant="outlined" 
                                            value={props.accountForm?.email}
                                            onKeyUp={props.accountKeyHandler}
                                            onChange={props.accountFormHandler}
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelShortBio}
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            multiline 
                                            minRows={6} 
                                            id="userAboutText"
                                            name="userAboutText" 
                                            variant="outlined" 
                                            value={props.accountForm?.userAboutText}
                                            onKeyUp={props.accountKeyHandler}
                                            onChange={props.accountFormHandler} 
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            <RenderText 
                                                {...props} 
                                                value={props.sectionAccountInformation?.labelIsActivated} 
                                            />
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <FormControlLabel
                                            control={<CustomSwitchStyle 
                                                name="checked" 
                                                checked={props.isUserActivated} 
                                                onChange={props.accountSwitchHandler} 
                                            />}
                                            label={props.sectionAccountInformation?.isActivatedText}
                                        />}
                                    </Grid>
                                </Grid>
                                <CustomDivider marginTop={5} marginBottom={2} />
                                <Grid className={classes.button_container_update}>
                                    <Box my={2}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="150px" height="40px" /> 
                                        : <UpdateAccountButton {...props} />}
                                    </Box>
                                </Grid>
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>
    );
}
