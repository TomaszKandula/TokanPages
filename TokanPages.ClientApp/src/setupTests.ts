// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
