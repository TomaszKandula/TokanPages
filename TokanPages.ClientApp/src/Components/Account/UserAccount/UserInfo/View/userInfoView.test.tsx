import "../../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { UserInfoView } from "./userInfoView";

describe("test account group component: userInfoView", () => {
    it("should render correctly '<UserInfoView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <UserInfoView
                    isLoading={false}
                    userStore={{
                        userId: "123456789",
                        isVerified: false,
                        aliasName: "Ester",
                        avatarName: "Ester.JPG",
                        firstName: "Ester",
                        lastName: "Exposito",
                        email: "exter.exposito@gmail.com",
                        shortBio: "Spanish Developer",
                        registered: "2020-01-01",
                        userToken: "123654789",
                        refreshToken: "951753789654123",
                        roles: [
                            {
                                id: "357159",
                                name: "Admin",
                                description: "System administrator",
                            },
                        ],
                        permissions: [
                            {
                                id: "951753",
                                name: "Access_All",
                            },
                        ],
                    }}
                    accountForm={{
                        firstName: "Ester",
                        lastName: "Exposito",
                        email: "ester.exposito@gmail.com",
                        userAboutText: "Spanish Developer",
                    }}
                    userImageName=""
                    isUserActivated={true}
                    isRequestingVerification={false}
                    formProgress={false}
                    keyHandler={jest.fn()}
                    formHandler={jest.fn()}
                    switchHandler={jest.fn()}
                    saveButtonHandler={jest.fn()}
                    verifyButtonHandler={jest.fn()}
                    sectionAccountInformation={{
                        caption: "USER DETAILS",
                        labelUserId: "User ID:",
                        labelEmailStatus: {
                            label: "Status",
                            negative: "Unverified",
                            positive: "Verified",
                        },
                        labelUserAlias: "User alias:",
                        labelFirstName: "First name:",
                        labelLastName: "last name:",
                        labelEmail: "Email:",
                        labelShortBio: "Bio:",
                        labelUserAvatar: "Avatar:",
                        updateButtonText: "Update",
                        uploadAvatarButtonText: "Upload",
                    }}
                    background={{ backgroundColor: "white" }}
                    userAbout={{
                        multiline: undefined,
                        minRows: undefined,
                    }}
                    fileUploadingCustomHandle="UserFile-Avatar"
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
