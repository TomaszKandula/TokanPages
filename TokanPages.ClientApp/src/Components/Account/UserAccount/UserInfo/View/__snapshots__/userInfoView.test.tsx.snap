// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: userInfoView should render correctly '<UserInfoView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="has-background-white"
      >
        <div
          class="bulma-container bulma-is-max-desktop"
        >
          <div
            class="py-6"
          >
            <div
              class="bulma-card"
            >
              <div
                class="bulma-card-content"
              >
                <p
                  class="is-size-4 has-text-grey"
                >
                  USER DETAILS
                </p>
                <hr />
                <div
                  class="py-4"
                >
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        User ID:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        123456789
                      </p>
                    </div>
                  </div>
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Status
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Unverified
                        <span>
                          Â (
                        </span>
                        <span
                          class="has-text-danger is-underlined has-pointer"
                        >
                          request verification
                        </span>
                        <span>
                          )
                        </span>
                      </p>
                    </div>
                  </div>
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        User alias:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Ester
                      </p>
                    </div>
                  </div>
                  <hr />
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth is-align-self-center"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Avatar:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <label
                        for="UserFile-Avatar"
                      >
                        <input
                          accept=""
                          hidden=""
                          id="UserFile-Avatar"
                          name="UserFile-Avatar"
                          type="file"
                        />
                        <button
                          class="icon-button button-upload is-flex is-align-self-center"
                          style="height: 48px; width: 48px;"
                        >
                          <div
                            class="avatar-wrapper user-avatar-avatar-small user-avatar-avatar-button"
                          >
                            <img
                              alt="User avatar"
                              class="avatar-image"
                              src="http://localhost:5000/api/v1/users/123456789/getUserImage?blobName="
                              title="Avatar"
                            />
                          </div>
                        </button>
                      </label>
                    </div>
                  </div>
                  <hr />
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth is-align-self-center"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        First name:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <div
                        class="bulma-control  "
                      >
                        <input
                          class="bulma-input   "
                          id="firstName"
                          name="firstName"
                          required=""
                          type="text"
                          value="Ester"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth is-align-self-center"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        last name:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <div
                        class="bulma-control  "
                      >
                        <input
                          class="bulma-input   "
                          id="lastName"
                          name="lastName"
                          required=""
                          type="text"
                          value="Exposito"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth is-align-self-center"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Email:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <div
                        class="bulma-control  "
                      >
                        <input
                          class="bulma-input   "
                          id="email"
                          name="email"
                          required=""
                          type="text"
                          value="ester.exposito@gmail.com"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="bulma-columns cancel-margin-bottom"
                  >
                    <div
                      class="bulma-column bulma-is-one-fifth is-align-self-center"
                    >
                      <p
                        class="is-size-6 has-text-grey"
                      >
                        Bio:
                      </p>
                    </div>
                    <div
                      class="bulma-column"
                    >
                      <div
                        class="bulma-field"
                      >
                        <div
                          class="bulma-control  "
                        >
                          <textarea
                            class="bulma-textarea   bulma-has-fixed-size"
                            id="userAboutText"
                            name="userAboutText"
                            required=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <hr />
                  <div
                    class="has-text-right"
                  >
                    <button
                      class="bulma-button bulma-is-info bulma-is-light"
                      type="submit"
                    >
                      Update
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="has-background-white"
    >
      <div
        class="bulma-container bulma-is-max-desktop"
      >
        <div
          class="py-6"
        >
          <div
            class="bulma-card"
          >
            <div
              class="bulma-card-content"
            >
              <p
                class="is-size-4 has-text-grey"
              >
                USER DETAILS
              </p>
              <hr />
              <div
                class="py-4"
              >
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      User ID:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      123456789
                    </p>
                  </div>
                </div>
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Status
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Unverified
                      <span>
                        Â (
                      </span>
                      <span
                        class="has-text-danger is-underlined has-pointer"
                      >
                        request verification
                      </span>
                      <span>
                        )
                      </span>
                    </p>
                  </div>
                </div>
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      User alias:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Ester
                    </p>
                  </div>
                </div>
                <hr />
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth is-align-self-center"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Avatar:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <label
                      for="UserFile-Avatar"
                    >
                      <input
                        accept=""
                        hidden=""
                        id="UserFile-Avatar"
                        name="UserFile-Avatar"
                        type="file"
                      />
                      <button
                        class="icon-button button-upload is-flex is-align-self-center"
                        style="height: 48px; width: 48px;"
                      >
                        <div
                          class="avatar-wrapper user-avatar-avatar-small user-avatar-avatar-button"
                        >
                          <img
                            alt="User avatar"
                            class="avatar-image"
                            src="http://localhost:5000/api/v1/users/123456789/getUserImage?blobName="
                            title="Avatar"
                          />
                        </div>
                      </button>
                    </label>
                  </div>
                </div>
                <hr />
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth is-align-self-center"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      First name:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <div
                      class="bulma-control  "
                    >
                      <input
                        class="bulma-input   "
                        id="firstName"
                        name="firstName"
                        required=""
                        type="text"
                        value="Ester"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth is-align-self-center"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      last name:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <div
                      class="bulma-control  "
                    >
                      <input
                        class="bulma-input   "
                        id="lastName"
                        name="lastName"
                        required=""
                        type="text"
                        value="Exposito"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth is-align-self-center"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Email:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <div
                      class="bulma-control  "
                    >
                      <input
                        class="bulma-input   "
                        id="email"
                        name="email"
                        required=""
                        type="text"
                        value="ester.exposito@gmail.com"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="bulma-columns cancel-margin-bottom"
                >
                  <div
                    class="bulma-column bulma-is-one-fifth is-align-self-center"
                  >
                    <p
                      class="is-size-6 has-text-grey"
                    >
                      Bio:
                    </p>
                  </div>
                  <div
                    class="bulma-column"
                  >
                    <div
                      class="bulma-field"
                    >
                      <div
                        class="bulma-control  "
                      >
                        <textarea
                          class="bulma-textarea   bulma-has-fixed-size"
                          id="userAboutText"
                          name="userAboutText"
                          required=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <hr />
                <div
                  class="has-text-right"
                >
                  <button
                    class="bulma-button bulma-is-info bulma-is-light"
                    type="submit"
                  >
                    Update
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
